runtime/vm/segment.c

```diff
@@ -130,10 +131,20 @@ void freeMemorySegment(J9JavaVM *javaVM, J9MemorySegment *segment, BOOLEAN freeD
 		} else if ((useAdvise) && (MEMORY_TYPE_JIT_SCRATCH_SPACE & segment->type)) {
 			j9mem_advise_and_free_memory(segment->baseAddress);
 		} else if (segment->type & (MEMORY_TYPE_RAM_CLASS | MEMORY_TYPE_UNDEAD_CLASS)) {
-			if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)) {
-				j9mem_free_memory32(segment->baseAddress);
+			if (IS_COLD_RUN(javaVM)) {
+				if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)) {
+					j9mem_free_memory32(segment->baseAddress); /* TODO: Create image memory free 32 bits */
```

Comment:
The initial prototype is only intended to work on 64bit mode (no-compreseedrefs) so support for `j9mem_free_memory32` is not required. 

In fact I would add an assertion here (`Assert_VM_unreachable()`) and other places where `J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)` is true on a cold run to make debugging easier.

Reply:
actually, something like `assert(!"There is no support for compressedRefs");` is probably better